app/src/main/java/pm/gnosis/heimdall/ui/safe/mnemonic/InputRecoveryPhraseActivity.kt

```diff
@@ -107,18 +107,18 @@ abstract class InputRecoveryPhraseActivity<VM : InputRecoveryPhraseContract> : V
                 noRecoveryNecessary(update.safeAddress)
             }
             is InputRecoveryPhraseContract.ViewUpdate.RecoverData -> {
-                onSuccess(update.executionInfo.transaction.wrapped.address)
+                onSuccess(update)
             }
         }
     }
 
     abstract fun noRecoveryNecessary(safe: Solidity.Address)
 
-    abstract fun onSuccess(safe: Solidity.Address)
+    abstract fun onSuccess(recoverData: InputRecoveryPhraseContract.ViewUpdate.RecoverData)
 
     companion object {
-        private const val EXTRA_SAFE_ADDRESS = "extra.string.safe_address"
-        private const val EXTRA_EXTENSION_ADDRESS = "extra.string.extension_address"
+        const val EXTRA_SAFE_ADDRESS = "extra.string.safe_address"
```

Comment:
why remove the private here?

Reply:
Used by the subclass `ReplaceBrowserExtensionRecoveryPhraseActivity`.

Accessing protected static members in the companion object of the super class is not supported yet so `protected` doesn't work.

Reply:
Yes! It was more to have both accessible to the subclasses if needed. Can make `EXTRA_SAFE_ADDRESS` private.